### ISSUE TITLE
Scanner finds comment in template string with placeholder

### ISSUE THREAD

**Issue by soda0289** [Microsoft/TypeScript#15115]
When using the scanner to find comments in typescript code a comment is detected inside of a template string that contains a placeholder. Remove the placeholder or changeing the template string to single/double quoted string fixes the issue. The scanner is user in the typescript-eslint-parser and was reported in this issue:
https://github.com/eslint/typescript-eslint-parser/issues/216

**TypeScript Version:**  2.2.1 / nightly (2.2.0-dev.201xxxxx)
Version 2.3.0-dev.20170318

**Code**
```ts
const ts = require("typescript");

var code = 'const a = `${folder}/**/*.ts`;';
var triviaScanner = ts.createScanner(ts.ScriptTarget.Latest, false, 0, code);

var kind = triviaScanner.scan();
while (kind !== ts.SyntaxKind.EndOfFileToken) {
    if (kind !== ts.SyntaxKind.SingleLineCommentTrivia && kind !== ts.SyntaxKind.MultiLineCommentTrivia) {
        kind = triviaScanner.scan();
        continue;
    }

    var range = {
        pos: triviaScanner.getTokenPos(),
        end: triviaScanner.getTextPos(),
        kind: triviaScanner.getToken()
    };

    var comment = code.substring(range.pos, range.end);
    console.log(comment);
    console.log(range);

    kind = triviaScanner.scan();
}
```

**Expected behavior:**
No comments found.

**Actual behavior:**
Finds comment in template string.

Output of script:
```
/**/
{ pos: 20, end: 24, kind: 3 }
```

**Comment by soda0289** [comment]
Sorry guys. Looks like there is no issue here. After reading the code I see there is a `reScanTemplateToken()` that will correctly scan template strings.
